### ISSUE TITLE
Strip optional falsy/nullish values from checked nodes

### ISSUE THREAD

**Issue by ottonomy** [openbadges/badgecheck#127]
Currently in badgecheck/tasks/validation.py validate_property():
```py
    if not required and (is_empty_list(values_to_test) or is_null_list(values_to_test)):
        return task_result(True, "Optional property {} is null in {} {}".format(
            prop_name, node_class, node_id
        ))
        # TODO Return STRIP_PROPERTY action
```

It would be nice in many cases to get consistent behavior that optional properties that are null, empty list, empty string etc. just get stripped out of the resultant graph.

We should get consensus on whether that is the right thing to do.

**Comment by cawmichael** [comment]
I agree that this kind of flexibility should be permitted for non-required properties.   We should allow for empty and null values including:

- null
- empty lists
- "" or even " " empty strings
